src/form-generator-multi-step.ts

```diff
@@ -33,19 +33,26 @@ export class FormGeneratorMultiStep {
         stepIndicators[step - 1].classList.add('current-step');
     }
 
-    createPage(title: string, fields: string[]): HTMLFormElement {
+    createPage = async (title: string, fields: string[]) => {
         const form = document.createElement('form') as HTMLFormElement;
         form.className = 'form-page';
         const pageTitle = document.createElement('h2');
         pageTitle.textContent = title;
         form.appendChild(pageTitle);
-    
-        fields.forEach((fieldName) => {
-            this.formElementCreator.createFormElement(form, this.config.fields.find((field: any) => field.name === fieldName));
-        });
 
         const formElementCreator = new FormElementCreator(new ElementCreator());
 
+        for (const fieldName of fields) {
+            const element = this.config.fields.find((field: any) => field.name === fieldName);
+            if (element) {
+                if (element.name === 'Phone') {
+                    await formElementCreator.createPhoneInput(form, element);
+                } else {
+                    formElementCreator.createFormElement(form, element);
```

Comment:
czym jest funkcja createForm element ? 
Co ona tworzy? jaki element? Bo brzmi ogolnie jakby miala tworzyc kazdy mozliwy a jednak telefonu nie potrafi zrobic :D bo masz do niej oddzielna funkcje. 
Poczytaj o strategii : 
https://refactoring.guru/design-patterns/strategy
I ogolnie o wzorcach projektowych. Zweryfikuj czy moze strategia miala by tu zastosowanie i sprobuj to zaimplementowac. 
Jezeli sie nie uda to bedziemy rozmawiac  o wzorcach projektowych na zajeciach :)